Project/core/Hotel.cs

```diff
@@ -1,88 +1,218 @@
-        private List<Reservation> cancelingReservations;
         private List<Reservation> reservations;
         private List<Room> rooms;
+        private List<Client> clients;
+        private List<Employee> employees;
+        private List<Log> logs;
+        private List<Account> accounts;
 
         // Konstruktory 
 
         public Hotel()
         {
             this.Name = null;
             this.reservations = new List<Reservation>();
-            this.cancelingReservations = new List<Reservation>();
             this.rooms = new List<Room>();
+            this.clients = new List<Client>();
+            this.employees = new List<Employee>();
+            this.logs = new List<Log>();
+            this.accounts = new List<Account>();
         }
 
-        public Hotel(string name) : this()
+        public Hotel(string name): this()
         {
-            this.Name = name;
+            this.Name =  name;
+            
         }
 
         // Getery i setery 
 
         public string Name { get => name; set => name = value; }
 
-        // Metody dodatkowe
+        internal List<Reservation> Reservations { get => reservations; set => reservations = value; }
+
+        internal List<Room> Rooms { get => rooms; set => rooms = value; }
+
+        internal List<Client> Clients { get => clients; set => clients = value; }
+
+        internal List<Employee> Employees { get => employees; set => employees = value; }
+
+        internal List<Log> Logs { get => logs; set => logs = value; }
 
-        public void CancelReservation(Reservation reservation)
+        internal List<Account> Accounts { get => accounts; set => accounts = value; }
+
+        // Metody dodatkowe => dotyczące rezerwacji 
+
+        public Reservation CreateReservation(string title, Client client, string checkInDate, string checkOutDate, int numberOfAdults, int numberOfChildren, int numberOfBabies, Employee employee, Room room)
         {
-            reservations.Remove(reservation);
-            cancelingReservations.Add(reservation);
+            Reservation reservation = new Reservation(title, client, checkInDate, checkOutDate, numberOfAdults, numberOfChildren, numberOfBabies, false, false, false);
+            reservations.Add(reservation);
+            reservation.AddRoom(room);
+            accounts.Find(a => a.Client == client).AddPayment(reservation);
+
+            string contents = "Create reservation: " + title;
+            AddLog(employee, Type.Create_Reserwation, contents);
+
+            return reservation;
         }
 
-        public void CancelReservationAfter2Days(Reservation reservation)
+        public bool ExtendReservations(DateTime date, Reservation reservation, Employee employee)
         {
-            CancelReservation(reservations.Find(r => r.CheckInDate.AddDays(2) == DateTime.Now && r.IsCheckIn == false));
+            string contents = "Extend reservation " + reservation.Title;
+            AddLog(employee, Type.Extend_Reservations, contents);
```

Comment:
Zawsze log powinien być na samym końcu funkcji, bo teraz log się doda nawet jak nie uda się przedłużyć rezerwacji.

(To samo masz w kilku innych miejscach w tej klasie)